docs(refactor): tutorials MTA-1809 (#443)
scaleway/docs-content: tutorials/foreman-puppet/index.mdx

tags: Foreman puppet Ubuntu-Focal puppet
categories:
  - compute
  - instances
dates:
  validation: 2022-02-28
  posted: 2018-08-06
---

## Foreman overview

[Foreman](https://www.theforeman.org/) is a tool that helps system administrators managing servers throughout their lifecycle, from provisioning and configuration to orchestration and monitoring. In short, it is a complete lifecycle management tool for physical and virtual servers. Foreman, available as open source software, becomes even more powerful when integrated with other open source projects such as [Puppet](/tutorials/configure-puppet/), [Chef](/tutorials/configure-chef-ubuntu-xenial/), [Salt](https://docs.saltstack.com/en/latest/), [Ansible](/tutorials/ansible-bionic-beaver/).

Foreman helps to automatize the OS installation. After that – through its very good integration with puppet – the new system will be configured towards the desired state. Finally Puppet will send facts about the system to Foreman which helps to monitor the whole system over its complete lifecycle. With a discovery plugin Foreman can also discover new machines in the network based on their mac address.

> This tutorial assumes that Foreman is being installed on a fresh instance, which will also act as the Puppet primary server. Foreman requires a working Puppet installation to perform the configuration management of hosts.

Reviewer: ```suggestion
This tutorial assumes that Foreman is being installed on a fresh Instance, which will also act as the Puppet primary server. Foreman requires a working Puppet installation to perform the configuration management of hosts.
```

> This tutorial assumes that Foreman is being installed on a fresh Instance, which will also act as the Puppet primary server. Foreman requires a working Puppet installation to perform the configuration management of hosts.

Reviewer 2: To me this should be a requirement

tags: Foreman puppet Ubuntu-Focal puppet
categories:
  - compute
  - instances
dates:
  validation: 2022-02-28
  posted: 2018-08-06
---

## Foreman overview

[Foreman](https://www.theforeman.org/) is a tool that helps system administrators managing servers throughout their lifecycle, from provisioning and configuration to orchestration and monitoring. In short, it is a complete lifecycle management tool for physical and virtual servers. Foreman, available as open source software, becomes even more powerful when integrated with other open source projects such as [Puppet](/tutorials/configure-puppet/), [Chef](/tutorials/configure-chef-ubuntu-xenial/), [Salt](https://docs.saltstack.com/en/latest/), [Ansible](/tutorials/ansible-bionic-beaver/).

Foreman helps to automatize the OS installation. After that – through its very good integration with puppet – the new system will be configured towards the desired state. Finally Puppet will send facts about the system to Foreman which helps to monitor the whole system over its complete lifecycle. With a discovery plugin Foreman can also discover new machines in the network based on their mac address.

> <<<<<<< HEAD

Author: ```suggestion
```

This tutorial assumes that Foreman is being installed on a fresh instance, which will also act as the Puppet primary server. 

> =======

Author: ```suggestion
```

> This tutorial assumes that Foreman is being installed on a fresh Instance, which will also act as the Puppet primary server. Foreman requires a working Puppet installation to perform the configuration management of hosts.

Author: ```suggestion
```

  ssh root@SERVER_IP
  ```

  To retrieve the Instances IP, list your exiting servers using `scw ps` (Scaleway CLI). For more information on the Scaleway CLI, refer to the tutorial on the [Scaleway Command Line Interface](https://github.com/scaleway/scaleway-cli/).

  The Instances IP can also be retrieved from the Scaleway Console. Once logged in, check the **IP Adresses** in the **Instances** tab of the left menu.

> 2. Check your Instances full hostname to ensure the hostname is set to a valid [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name).

Reviewer: ```suggestion
2. Check your Instance's full hostname to ensure the hostname is set to a valid [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name).
```

    * Foreman is running at https://foreman.example.com
        Initial credentials are admin / 6fHtmSWpqNbGo7nn
    * Foreman Proxy is running at https://foreman.example.com:8443
    * Puppetprimary is running at port 8140
    The full log is at /var/log/foreman-installer/foreman.log
  ```

## Managing Puppet

> During installation, the installation tool will sets up a puppet primary on the host, which is fully integrated with Foreman. Run the Puppet agent on the instance to send the first Puppet report to Foreman, This creates the host automatically in Foremans database.

Reviewer: ```suggestion
During installation, the installation tool will sets up a puppet primary on the host, which is fully integrated with Foreman. Run the Puppet agent on the Instance to send the first Puppet report to Foreman, This creates the host automatically in Foremans database.
```

> During installation, the installation tool will sets up a puppet primary on the host, which is fully integrated with Foreman. Run the Puppet agent on the Instance to send the first Puppet report to Foreman, This creates the host automatically in Foremans database.

Reviewer 2: ```suggestion
During installation, the installation wizard sets up a puppet primary on the host, which is fully integrated with Foreman. Run the Puppet agent on the Instance to send the first Puppet report to Foreman, This creates the host automatically in Foremans database.
```
